common/Util.hpp

```diff
@@ -303,7 +303,9 @@ namespace Util
             if (data[i] >> 6 != 0x3)
                 return i;
             int chunkLen = 1;
-            for (; data[i] & (1 << (7-chunkLen)); chunkLen++);
+            for (; data[i] & (1 << (7-chunkLen)); chunkLen++)
+                if (chunkLen > 4)
+                    return i;
             for (; chunkLen > 1; --chunkLen)
                 if (data[++i] >> 6 != 0x2)
```

Comment:
heh - we should also check that i + chunkLen < len after we've worked out the chunkLen.
my bad ! thanks for catching; can you tweak ? =)